crates/forge_loader/src/manifest.rs

```diff
@@ -25,24 +25,54 @@ pub struct FunctionMod<'a> {
     providers: Option<AuthProviders<'a>>,
 }
 
+// Abstracting away key, function, and resolver into a single struct for reuse whoo!
 #[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
-struct ModInfo<'a> {
+struct CommonKey<'a> {
     key: &'a str,
-    title: &'a str,
+    function: &'a str,
+    resolver: Option<&'a str>,
```

Comment:
<!--
 Copyright 2024 Joshua Wong.
 SPDX-License-Identifier: Apache-2.0
-->

This deserialization is incorrect, `resolver` should be an `Option<Resolver<'a>>` where `Resolver` is defined like so:

```rust
#[derive(Debug, Default, Clone, PartialEq, Eq, Deserialize)]
struct Resolver<'a> {
    function: Option<&'a str>,
}
```

See <https://developer.atlassian.com/platform/forge/build-a-custom-ui-app-in-confluence/#change-the-content-byline-item-title> as an example of how the `resolver` property should look.

Also, `function` should be an `Option<&'a str>`, since `Forge` modules do not have to specify a `function` if they specify a `resolver`. Otherwise, we'll error out whenever someone uses a module with a `resolver` and no `function`.